src/compiler.rkt

```diff
@@ -482,7 +527,7 @@
                                    zhi-exp)))
      (max 0
           (- condition-lhs (min outlo-exp outhi-exp)
-             (if (xor (bigfloat-signbit outhi) (bigfloat-signbit outlo)) ; cancellation when output crosses 0
+             (if (equal? (bigfloat-signbit outhi) (bigfloat-signbit outlo)) ; cancellation when output crosses 0
```

Comment:
Ok so this changes two things. First, instead of always false, it's now sometimes true. Doesn't this also invert the condition? Like, `xor` means `!=`, the opposite of `==`.

Reply:
Did we have two bugs?

Reply:
Yeah, the my implementation was completely wrong, firstly, xor didn't work, secondly, even the order was incorrect. I was always checking myself in terminal as `(xor 0 1)`, which is false for all integers turned out. Now it should be correct